src/libraries/System.Text.RegularExpressions/gen/Polyfills.cs

```diff
@@ -0,0 +1,84 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Buffers;
+using System.Collections;
+using System.Collections.Generic;
+using System.Text;
+
+// This file provides helpers used to help compile some Regex source code (e.g. RegexParser) as part of the netstandard2.0 generator assembly.
+
+namespace System.Text
+{
+    internal static class StringBuilderExtensions
+    {
+        public static StringBuilder Append(this StringBuilder stringBuilder, ReadOnlySpan<char> span) =>
+            stringBuilder.Append(span.ToString());
+
+        public static ReadOnlyMemory<char>[] GetChunks(this StringBuilder stringBuilder)
+        {
+            var chars = new char[stringBuilder.Length];
+            stringBuilder.CopyTo(0, chars, 0, chars.Length);
+            return new[] { new ReadOnlyMemory<char>(chars) };
+        }
+    }
+}
+
+namespace System
+{
+    internal static class StringExtensions
+    {
+        public static string Create<TState>(int length, TState state, SpanAction<char, TState> action)
+        {
+            var array = new char[length];
+            action(array, state);
+            return new string(array);
+        }
+    }
+}
+
+namespace System.Buffers
+{
+    internal delegate void SpanAction<T, in TArg>(Span<T> span, TArg arg);
+}
+
+namespace System.Threading
+{
+    internal static class InterlockedExtensions
+    {
+        public static int Or(ref int location1, int value)
+        {
+            int current = location1;
+            while (true)
+            {
+                int newValue = current | value;
+                int oldValue = Interlocked.CompareExchange(ref location1, newValue, current);
+                if (oldValue == current)
+                {
+                    return oldValue;
+                }
+                current = oldValue;
+            }
+        }
+    }
+}
+
+namespace System.Text.RegularExpressions
+{
+    internal class RegexReplacement
```

Comment:
I think the rest of the code in this file makes sense, but this one kind of surprised me. It doesn't feel like the same category as the rest, since we are compiling the rest of the Regex code into this assembly. This one is more like a "stub" than a "polyfill" right?

Reply:
I guess.  I can rename the file to Stubs.cs if that addresses your concern.  This is "make the rest of the code compile and work correctly".cs.

Reply:
I think just a simple comment on this class would have alleviated my "surprise". The rest of them all made sense - new APIs that aren't available in netstandard2.0. This one just caused an "interrupt" in my mental model of this file.